nemo_skills/inference/model/__init__.py

```diff
@@ -53,6 +53,9 @@
 def get_model(server_type, tokenizer=None, **kwargs):
     """A helper function to make it easier to set server through cmd."""
     model_class = models[server_type.lower()]
+    if server_type == "trtllm" and kwargs.get("enable_soft_fail", False):
+        if kwargs.get("context_limit_retry_strategy", None) is not None:
+            raise ValueError("context_limit_retry_strategy is not supported for trtllm")
```

Comment:
what about api models, are they supported or it's case-by-case there?

Reply:
I have not tested API models so far.
It's a manual process where you get the errors: 
- for exceeding the length with the prompt only and 
- for exceeding the length with the generation messages + prompt.
 
Add parts of those error messages that we can compare against. And then write a regex to extract the error info from that. 

Let me add it in another PR.